alacritty/src/display/content.rs

```diff
@@ -222,18 +222,21 @@ impl RenderableCell {
         let viewport_start = Point::new(Line(-(display_offset as i32)), Column(0));
         let colors = &content.config.colors;
         let mut character = cell.c;
-
-        if let Some((c, is_first)) =
-            content.hint.as_mut().and_then(|hint| hint.advance(viewport_start, cell.point))
-        {
-            let (config_fg, config_bg) = if is_first {
-                (colors.hints.start.foreground, colors.hints.start.background)
+        let mut flags = cell.flags;
+        if content.hint.as_mut().map(|hint| hint.matches.advance(cell.point)).unwrap_or(false) {
+            if let Some((c, is_first)) =
+                content.hint.as_mut().and_then(|hint| hint.advance(viewport_start, cell.point))
+            {
+                let (config_fg, config_bg) = if is_first {
+                    (colors.hints.start.foreground, colors.hints.start.background)
+                } else {
+                    (colors.hints.end.foreground, colors.hints.end.background)
+                };
+                character = c;
+                Self::compute_cell_rgb(&mut fg, &mut bg, &mut bg_alpha, config_fg, config_bg);
```

Comment:
```suggestion
                Self::compute_cell_rgb(&mut fg, &mut bg, &mut bg_alpha, config_fg, config_bg);
                
                character = c;
```

Doesn't make sense to interrupt the color determination/computation with the character assignment when the character doesn't depend on the color determination.